src/index.ts

```diff
@@ -1,233 +1,328 @@
-import Serverless from "serverless";
-import { Options } from "serverless";
+import Serverless from 'serverless';
+import { Options } from 'serverless';
 import chalk from 'chalk';
-import { camelCase, paramCase as kebabCase } from "change-case";
-import Aws from "serverless/plugins/aws/provider/awsProvider";
-import Service from "serverless/classes/Service";
+import { camelCase, paramCase as kebabCase } from 'change-case';
+import Aws from 'serverless/plugins/aws/provider/awsProvider';
+import Service from 'serverless/classes/Service';
 
 type ConventionsConfig = {
-     ignore?: {
-          serviceName?: boolean,
-          handlerName?: boolean,
-          functionName?: boolean,
-          handlerNameMatchesFunction?: boolean,
-          dynamoDBTableName?: boolean,
-     }
-}
+  ignore: {
```

Comment:
We want `ignore` to be optional, don't we?

Reply:
Yep, should be optional

Reply:
I though it should be optional too but then at the end of constructor, we did something like this:
```
// Make sure ignore is defined to prevent errors being from being thrown when referencing children
    if (this.conventionsConfig === undefined) {
      this.conventionsConfig = {};
    }
    if (this.conventionsConfig.ignore === undefined) {
      this.conventionsConfig.ignore = {};
    }
```
Therefore, I think ignore should not be optional. Only its children are optional.

Reply:
Been a while since I wrote this so trying to remember :sweat_smile: 

I think this is because in the `serverless.yml` file if you don't specify `ignore` it comes through as undefined. Therefore the `ignore` variable should be optional.

Reply:
I had a discussion with Torb the other day about this. We might need to double check this later for few different cases:

- `conventions` is not specified in `serverless.yml`
- `conventions` is specified in `serverless.yml` without `ignore`
- `conventions` is specified in `serverless.yml` with empty `ignore`

For now, let me just revert this change to make this PR focus to adding stage restriction. Once we're happy with this, I'll revisit `ignore` later.